### PR TITLE
[improve][broker] Add an error log to troubleshoot the failure of starting broker registry.

### PR DESCRIPTION
**Pull request by liangyepianzhou**
### Motivation
When the `ExtensibleLoadManagerImpl.java` fails to start, it will close `brokerRegistry`. But we cannot directly determine why the broker register is closed. 

```log
2024-02-15T20:09:18,675+0000 [main] ERROR org.apache.pulsar.broker.namespace.NamespaceService - java.lang.IllegalStateException: The registry already closed.
java.util.concurrent.ExecutionException: java.lang.IllegalStateException: The registry already closed.
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2073) ~[?:?]
	at org.apache.pulsar.broker.namespace.NamespaceService.registerNamespace(NamespaceService.java:380) ~[io.streamnative-pulsar-broker-3.1.2.1.jar:3.1.2.1]
	at org.apache.pulsar.broker.namespace.NamespaceService.registerBootstrapNamespaces(NamespaceService.java:343) ~[io.streamnative-pulsar-broker-3.1.2.1.jar:3.1.2.1]
	at org.apache.pulsar.broker.PulsarService.start(PulsarService.java:867) ~[io.streamnative-pulsar-broker-3.1.2.1.jar:3.1.2.1]
	at org.apache.pulsar.PulsarBrokerStarter$BrokerStarter.start(PulsarBrokerStarter.java:276) ~[io.streamnative-pulsar-broker-3.1.2.1.jar:3.1.2.1]
	at org.apache.pulsar.PulsarBrokerStarter.main(PulsarBrokerStarter.java:356) ~[io.streamnative-pulsar-broker-3.1.2.1.jar:3.1.2.1]
Caused by: java.lang.IllegalStateException: The registry already closed.
```

### Modifications
Add an error log when the `ExtensibleLoadManagerImpl.java` fails to start,

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java [hunk 1]
```diff
@@ -444,7 +444,9 @@ public void start() throws PulsarServerException {
             this.initWaiter.countDown();
             this.started = true;
         } catch (Exception ex) {
+            log.error("Failed to start the extensible load balance.", ex);
```
Comment (poorbarcode) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:447]: How about merging these two logs into one?

**Review by poorbarcode — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java (hunk 1 above)
Comment (congbobo184) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java:447]: +1

Comment (liangyepianzhou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/ExtensibleLoadManagerImpl.java]: Thanks for the suggestion. Merged.

**Review by congbobo184 — APPROVED** [review]

**Review by Demogorgon314 — APPROVED** [review]
LGTM

**Review by nodece — APPROVED** [review]

**Pull request merged**